src/transaction.ts

```diff
@@ -144,6 +155,10 @@ export interface RunUpdateCallback {
 }
 
 export type CommitCallback = NormalCallback<spannerClient.spanner.v1.ICommitResponse>;
+export interface BeginTransactionOptions {
+  gaxOptions?: CallOptions;
+  transactionRequestOptions?: Pick<TxnRequestOptions, 'transactionTag'>;
```

Comment:
`BeginTransaction` should not be tagged. From the spec:

> All requests sent using the transaction (excluding BeginTransaction) should include the
transaction tag regardless of whether a request tag has been set.

That would probably also remove the requirement to pass in any transaction options to `getWriteSession`.

Reply:
> If the library supports hand-crafted transactions, creating the transaction should support setting
> a transaction tag so that it may be used in the BeginTransaction request.

Reply:
Otherwise transactions are created at the time `database` is created without transaction tags, so in that case no transaction tag will be passed to `Transaction#begin()`

Reply:
> > If the library supports hand-crafted transactions, creating the transaction should support setting
> > a transaction tag so that it may be used in the BeginTransaction request.

Sorry, I missed that one. So in case of a hand-crafted read/write transaction setting a transaction tag on the `BeginTransaction` request should be supported. This still means that we should not allow this for read-only transactions, which also use this method (and most probably is the most common reason for applications to use this method).


> Otherwise transactions are created at the time `database` is created without transaction tags, so in that case no transaction tag will be passed to `Transaction#begin()`

Correct, which is also one of the reasons why I would say that `getWriteSession()` should never support setting a transaction tag:
1. There is no guarantee that `getWriteSession()` will invoke `BeginTransaction`. If it finds a session that already has a read/write transaction in the pool, it will return that session. The transaction on that session has not been tagged.
2. There is the possibility that `getWriteSession()` does invoke `BeginTransaction` and returns the transaction that is created by that call. That transaction will include the tag.

So the result of calling `getWriteSession()` with a transaction tag is non-deterministic, which will be confusing.